Makefile

```diff
@@ -67,51 +63,90 @@ CI_DOCKER_CACHE_DIR := $(CI_CACHE_DIR)/docker
 CI_DOCKER_IMAGE := yarpc_go
 CI_DOCKER_CACHE_FILE := $(CI_DOCKER_CACHE_DIR)/$(CI_DOCKER_IMAGE)
 
-DOCKER_COMPOSE_VERSION := 1.10.0
-
-_BIN_DIR = $(CI_CACHE_DIR)/bin
-
-$(_BIN_DIR)/docker-compose:
-	mkdir -p $(_BIN_DIR)
-	curl -L https://github.com/docker/compose/releases/download/$(DOCKER_COMPOSE_VERSION)/docker-compose-$(shell uname -s)-$(shell uname -m) > $(_BIN_DIR)/docker-compose
-	chmod +x $(_BIN_DIR)/docker-compose
+TMP := $(shell pwd)/.tmp
 
-##############################################################################
-
-_GENERATE_DEPS_DIR = $(shell pwd)/.tmp
-$(_GENERATE_DEPS_DIR):
-	mkdir $(_GENERATE_DEPS_DIR)
-
-# Full paths to executables needed for 'make generate'
-_GENERATE_DEPS_EXECUTABLES = $(_GENERATE_DEPS_DIR)/thriftrw-plugin-yarpc $(_GENERATE_DEPS_DIR)/thrift
-
-# Special-case for local executables
-$(_GENERATE_DEPS_DIR)/thriftrw-plugin-yarpc: ./encoding/thrift/thriftrw-plugin-yarpc/*.go $(_GENERATE_DEPS_DIR)
-	go build -o $(_GENERATE_DEPS_DIR)/thriftrw-plugin-yarpc ./encoding/thrift/thriftrw-plugin-yarpc
-
-$(_GENERATE_DEPS_DIR)/thrift:
-	./scripts/install-thrift.sh $(_GENERATE_DEPS_DIR)
+# all we want is go get -u github.com/Masterminds/glide
+# but have to pin to 0.12.3 due to https://github.com/Masterminds/glide/issues/745
+GLIDE_VERSION := 0.12.3
+DOCKER_COMPOSE_VERSION := 1.10.0
+THRIFT_VERSION := 1.0.0-dev
+
+BIN = $(CI_CACHE_DIR)/bin
+
+$(BIN)/docker-compose:
+	mkdir -p $(BIN)
+	curl -L https://github.com/docker/compose/releases/download/$(DOCKER_COMPOSE_VERSION)/docker-compose-$(shell uname -s)-$(shell uname -m) > $(BIN)/docker-compose
+	chmod +x $(BIN)/docker-compose
+
+$(BIN)/glide:
+	mkdir -p $(BIN)
+	mkdir -p $(TMP)/glide
+	curl -L https://github.com/Masterminds/glide/releases/download/v$(GLIDE_VERSION)/glide-v$(GLIDE_VERSION)-$(shell uname -s)-amd64.tar.gz > $(TMP)/glide/glide.tar.gz
+	tar -C $(TMP)/glide -xzf $(TMP)/glide/glide.tar.gz
+	mv $(TMP)/glide/$(shell uname -s | tr '[:upper:]' '[:lower:]')-amd64/glide $(BIN)/glide
+	rm -rf $(TMP)/glide
+
+$(BIN)/thrift:
+	mkdir -p $(BIN)
+	mkdir -p $(TMP)/thrift
+	curl -L "https://github.com/uber/tchannel-go/releases/download/thrift-v$(THRIFT_VERSION)/thrift-1-$(shell uname -s)-$(shell uname -m).tar.gz" > $(TMP)/thrift/thrift.tar.gz
+	tar -C $(TMP)/thrift -xzf $(TMP)/thrift/thrift.tar.gz
+	mv $(TMP)/thrift/thrift-1 $(BIN)/thrift
+	rm -rf $(TMP)/thrift
+
+$(BIN)/thriftrw-plugin-yarpc: ./encoding/thrift/thriftrw-plugin-yarpc/*.go
+	mkdir -p $(BIN)
+	go build -o $(BIN)/thriftrw-plugin-yarpc ./encoding/thrift/thriftrw-plugin-yarpc
+
+DOCKER_COMPOSE = $(BIN)/docker-compose
+GLIDE = $(BIN)/glide
+THRIFT = $(BIN)/thrift
+BINS = $(DOCKER_COMPOSE) $(GLIDE) $(THRIFT) $(BIN)/thriftrw-plugin-yarpc
```

Comment:
Little nit: Why not declare `GLIDE`, `DOCKER_COMPOSE`, etc. before the rules to build them and use `$(GLIDE)` as the rule target?

```
GLIDE = $(BIN)/glide

$(GLIDE):
    mkdir -p $(BIN)
    ...
```